homeassistant/components/remootio/config_flow.py

```diff
@@ -0,0 +1,182 @@
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the class."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        user_input = user_input or {}
+        errors = {}
+
+        if len(user_input) != 0:
+            validation_result = {}
+
+            try:
+                validation_result = await validate_input(self.hass, user_input)
+            except UnsupportedRemootioDeviceError:
+                _LOGGER.debug("Remootio device isn't supported", exc_info=True)
+                return self.async_abort(reason="unsupported_device")
+            except vol.MultipleInvalid as ex:
+                _LOGGER.error(
+                    "Invalid user input. MultipleInvalid.Errors [%s]", ex.errors
+                )
+                for error in ex.errors:
+                    _LOGGER.debug(
+                        "Error [%s] Path [%s]", error.__class__.__name__, error.path[0]
+                    )
+                    if isinstance(error, RequiredFieldInvalid):
+                        errors[str(error.path[0])] = f"{error.path[0]}_required"
+                    else:
+                        errors[str(error.path[0])] = f"{error.path[0]}_invalid"
+            except RemootioClientConnectionEstablishmentError:
+                _LOGGER.error("Can't connect to Remootio device")
+                errors["base"] = "cannot_connect"
+            except RemootioClientAuthenticationError:
+                _LOGGER.error("Can't authenticate by the Remootio device")
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception/error")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    validation_result[CONF_DATA][CONF_SERIAL_NUMBER]
+                )
+                self._abort_if_unique_id_configured(validation_result[CONF_DATA])
+
+                return self.async_create_entry(
+                    title=validation_result[CONF_TITLE],
+                    data=validation_result[CONF_DATA],
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_HOST,
+                        default=user_input.get(CONF_HOST, vol.UNDEFINED),
+                    ): vol.Coerce(str),
+                    vol.Optional(
+                        CONF_API_SECRET_KEY,
+                        default=user_input.get(CONF_API_SECRET_KEY, vol.UNDEFINED),
+                    ): vol.Coerce(str),
+                    vol.Optional(
+                        CONF_API_AUTH_KEY,
+                        default=user_input.get(CONF_API_AUTH_KEY, vol.UNDEFINED),
+                    ): vol.Coerce(str),
```

Comment:
Why are they all optional?

Reply:
Because the `user_input` will be validated based on the `INPUT_VALIDATION_SCHEMA` by invoking the method `validate_input` in line 114 and if a validation error occurs, an meaningful message will be shown to the user.

Reply:
The user should not have to look at error messages in the log to understand how to fill in the schema, please make the fields required here as requested by @joostlek